src/dynamic_honey_badger/error.rs

```diff
@@ -92,3 +92,51 @@ impl Display for Error {
 }
 
 pub type Result<T> = ::std::result::Result<T, Error>;
+
+/// Represents each way an an incoming message can be considered faulty.
+#[derive(Debug, Fail, PartialEq)]
+pub enum FaultKind {
```

Comment:
Not sure whether it should go into `error` or a separate `fault` submodule (same in the other algorithms). Let's see what the other reviewers say.

Reply:
I don't think it should be in the `error` module, as faults are not errors. I'd just put it in `mod.rs`.